docs/Design/Loops/For Each Pair.md

```diff
@@ -0,0 +1,199 @@
+# For Each Pair
+
+A loop that iterates over a dictionary by pair.
+
+## Commands
+
+### `for each pair start`
+
+`for each pair start : container pair key string value int`
+
+Starts a for each pair loop.
+The first parameter is the container to be iterated over.
+The second parameter is the iterator variable, of type pair.
+The third parameter is the variable representing the key portion of the pair, of type specified by the fourth parameter.
+The fifth parameter is the variable representing the value portion of the pair, of type specified by the sixth parameter.
+
+## Usage
+```
+for each pair start : container pair key string value int
+    comment line : ...
+for each end
+```
+
+### CSharp
+```csharp
+foreach (KeyValuePair<string, int> pair in container)
+{
+    string key = pair.Key;
+    int value = pair.Value;
+    // ...
+}
+```
+
+### Java
+```java
+for (Map.Entry<string, int> pair : container.entrySet()) {
+    string key = pair.getKey();
+    int value = pair.getValue();
+    // ...
+}
+```
+
+### Python
+```python
+for key, value in container.iteritems():
+    # ...
+```
+
+### Ruby
+```ruby
+container.each { |key, value|
+    # ...
+}
+```
+
+### TypeScript
+```typescript
+for (let key: string in container) {
```

Comment:
Still not valid TypeScript.

Does GLS currently output this? If so, it's a bug.